src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -389,22 +396,25 @@ public boolean updateRepairSegment(RepairSegment newRepairSegment) {
 
   @Override
   public Collection<RepairSegment> getRepairSegmentsForRun(long runId) {
-    List<ResultSetFuture> segmentsFuture = Lists.<ResultSetFuture>newArrayList();
-    Collection<RepairSegment> segments = Lists.<RepairSegment>newArrayList();
+    List<ResultSetFuture> segmentsFuture = Lists.newArrayList();
+    Collection<RepairSegment> segments = Lists.newArrayList();
 
     // First gather segments ids
     ResultSet segmentsIdResultSet = session.execute(getRepairSegmentByRunIdPrepStmt.bind(runId));
-    for(Row segmentIdResult:segmentsIdResultSet){
+    int i=0;
+    for(Row segmentIdResult:segmentsIdResultSet) {
       // Then get segments by id
       segmentsFuture.add(session.executeAsync(getRepairSegmentPrepStmt.bind(segmentIdResult.getLong("segment_id"))));
-
-      for(ResultSetFuture segmentResult:segmentsFuture){
-        Row segmentRow = segmentResult.getUninterruptibly().one();
-        if(segmentRow!=null){
-          segments.add(createRepairSegmentFromRow(segmentRow));
-        }
+      i++;
+      if(i%100==0 || segmentsIdResultSet.isFullyFetched()) {
+        for(ResultSetFuture segmentResult:segmentsFuture) {
+          Row segmentRow = segmentResult.getUninterruptibly().one();
+          if(segmentRow!=null){
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)